src/test/suite/alphaInstaller.test.ts

```diff
@@ -0,0 +1,123 @@
+        isCapabilityEnabled = sinon.stub(capabilities, "isCapabilityEnabled");
+        installCommand = sinon.stub(vscode.commands, 'executeCommand');
+        version = sinon.stub(vscode, 'version')
+        installedVersion = sinon.stub(context.tabnineContext, 'version')
+        httpMock = sinon.stub(https, 'request');
+        tmpMock = sinon.stub(tmp, 'file');
+        createWriteStreamMock = sinon.stub(fs, 'createWriteStream');
+        createWriteStreamMock.returns(new PassThrough())
+    });
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    test("in case of not alpha, do nothing", async () => {
+        await runInstallation("3.0.11-alpha", "v3.1.11", '1.32.0', false)
+
+        assertWasNotInstalled();
+    });
+    test("in case of alpha and unsupported vscode api(1.35), do nothing", async () => {
+        await runInstallation("3.0.11-alpha", "v3.1.11", '1.32.0')
+        assertWasNotInstalled();
+    });
+
+    ['3.1.10', '3.0.10'].forEach((installed) => {
+        test(`in case of TabNine released version is lower or equal to current version (${installed}), do nothing`, async () => {
+            await runInstallation(installed, "v3.0.10")
+            assertWasNotInstalled();
+        })
+    })
+    test("in case of newer GA version, do nothing", async () => {
+        await runInstallation("3.0.11-alpha", "v3.1.11");
+        assertWasNotInstalled();
+    })
+
+    test("in case of newer alpha version and current GA should update", async () => {
+        await runInstallation("3.0.11", "v3.1.11-alpha");
+        assertSuccessfulInstalled();
+    })
+
+    test("in case of newer alpha version, install the new one", async () => {
+        await runInstallation("3.1.10-alpha.150", "v3.1.10-alpha.280345345");
+        assertSuccessfulInstalled();
+    })
+
+    async function runInstallation(installed: string, available: string, vscodeVersion = minimalSupportedVscodeVersion, isAlpha = true){
+        setIsAlpha(isAlpha);
+        version.value(vscodeVersion);
+        installedVersion.value(installed);
+        const artifactUrl = getArtifactUrl(available);
+
+        mockRequest({ assets: [{ browser_download_url: artifactUrl}] }, latestReleaseUrl);
+        mockRequest({ data: "test" }, artifactUrl);
+
+        mockTempFile();
+
+        return handleAlpha();
+    }
+
+    function mockTempFile(){
+        tmpMock.yields(null, tempFileName, null)
+    }
+
+    function mockRequest(data: unknown, url: string) {
+        const streamMock: PassThrough & {statusCode?: number} = new PassThrough();
+        streamMock.push(JSON.stringify(data));
+        streamMock.end();
+        streamMock.statusCode = 200;
+
+        httpMock.withArgs(url)
+            .callsFake((_url, callback: ( stream: PassThrough & {statusCode?: number}) => void) => {
+                callback(streamMock);
+                return { end: sinon.stub(), on: sinon.stub() };
+            })
+    }
+    function assertWasNotInstalled() {
+        assert(
```

Comment:
move any function that is not a test to a different file (I've used the convension `utils/testname.utils.ts`)